tdrs-backend/tdpservice/parsers/parse.py

```diff
@@ -61,27 +41,69 @@ def parse_datafile(datafile):
         errors['document'] = ['Section does not match.']
         return errors
 
-    line_errors = parse_datafile_lines(datafile, program_type, section)
+    line_errors = parse_datafile_lines(rawfile, program_type, section)
 
     errors = errors | line_errors
 
     return errors
 
 
-def parse_datafile_lines(datafile, program_type, section):
+def store_record(unsaved_records, record, model):
+    """Store record in dictionary for later processing."""
+    if record:
+        if model not in unsaved_records:
+            unsaved_records[model] = [record]
+        else:
+            unsaved_records[model].append(record)
```

Comment:
But I see here that unsaved_records is not dictionary??

Reply:
unsaved records is a dict, but its keys point to a list of records to be serialized. I can split that up to make it less confusing.